File: CS/CheckedComboBoxBoundToCollection/Form1.cs
using System;
using DevExpress.XtraEditors;
using System.Collections.Generic;

namespace CheckedComboBoxBoundToCollection {
    public partial class Form1 :XtraForm {
        int cnt;

        public Form1() {
            InitializeComponent();
        }

        private void OnAddOrderButtonClick(object sender, EventArgs e) {
            ordersSource.Add(new Order(string.Concat("Order ", ++cnt)));
        }
    }
}

File: VB/CheckedComboBoxBoundToCollection/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.XtraEditors
Imports System.Collections.Generic

Namespace CheckedComboBoxBoundToCollection
	Partial Public Class Form1
		Inherits XtraForm
		Private cnt As Integer

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub OnAddOrderButtonClick(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
			cnt += 1
			ordersSource.Add(New Order(String.Concat("Order ", cnt)))
		End Sub
	End Class
End Namespace